src/functions.js

```diff
@@ -121,7 +120,27 @@ var _ = Mavo.Functions = {
 			return n && n > 1? [arr] : arr;
 		}
 
-		return n && n > 1? arr.slice(0, n) : arr[0];
+		if (n && n > 1) {
+			var ret = [];
+
+			for (i = 0; i<arr.length && ret.length<n; i++) {
+
+				if (Mavo.value(arr[i]) !== null) {
+					ret.push(arr[i]);
+				}
+
+			}
+
+			return ret;
+		} 
+		else {
+
+			for (i = 0; i<arr.length && Mavo.value(arr[i]) == null; i++) {
```

Comment:
Avoid the `==` operator unless you have a good reason to use it. Use `===` instead.
There is a lot of weirdness around it due to the way it does type coercion, and is the root of most JSWTFs. 
This is a fantastic article about type coercion: https://medium.freecodecamp.org/js-type-coercion-explained-27ba3d9a2839